### PR TITLE
Update cortex-m, cortex-m-rt, stm32f103xx

### PR DESCRIPTION
**Pull request by ykomatsu**
`cortex-m-rtfm 0.3.4` is not compatible with latest `stm32f103xx-hal` (>= japaric/stm32f103xx-hal@5487385a8d05c5b8ea28e59b18a627d1acaefacf)

Usage:
1. Add ~~`cortex-m-rt-macros` and~~ `panic-` *something* crate to `[dependencies]` table in your `Cargo.toml`
2. Add `#![no_main]` attribute to your code
3. Add ~~`extern crate cortex-m-rt-macros;` and~~ `extern crate panic_` *something* `;` to your code

### CODE REVIEW

File: examples/zero-tasks.rs [hunk 1]
```diff
@@ -2,8 +2,11 @@
 #![deny(unsafe_code)]
 #![deny(warnings)]
 #![no_std]
+#![no_main]
 
+extern crate cortex_m_rt_macros;
```
Comment (TeXitoi) [inline examples/zero-tasks.rs]: That would be better to depend on cortex_m_rt as you already need it on your Cargo.toml to use the rt feature.

Comment (TeXitoi) [reply examples/zero-tasks.rs]: I think the better solution would be to reexport the needed macro directly in cortex-m-rtfm to not depend on extern crate by the user.

Comment (ykomatsu) [reply examples/zero-tasks.rs]: I think you're right.

Could you check 0054ef7c387c911c168f103ba59ae3d4751a2201, please?

Comment (TeXitoi) [reply examples/zero-tasks.rs]: Great! https://github.com/TeXitoi/rusty-clock/commit/1f10a38bc173f3661f6545afba71741dfd2b62c6

File: macros/src/trans.rs [hunk 2]
```diff
@@ -18,8 +19,15 @@ pub fn app(app: &App, ownerships: &Ownerships) -> TokenStream {
...
+        use #krate::entry;
+
+        #[entry]
+        fn main() -> ! {
+            main_impl();
+        }
+
         #[allow(unsafe_code)]
-        fn main() {
+        fn main_impl() -> ! {
```
Comment (TeXitoi) [inline macros/src/trans.rs]: Can't you just add `#[entry]` on main without creating `main_impl`?

Comment (ykomatsu) [reply macros/src/trans.rs]: Well, let's see, is this all right? 493f71943a0d983d636f89135a88dd141b5e80d5

**Comment by ReeceStevens** [comment]
I would like to chime in for this PR; the current cortex-m-rtfm crate is actually not binding exception handlers at the moment due to the `SYS_TICK` -> `SysTick` change and it has broken my application. Happy to jump in if there's any extra help needed here-- @TeXitoi or @ykomatsu , do either of you know the state of this?

**Comment by ykomatsu** [comment]
@ReeceStevens Thank you for checking my branch!

@japaric If there is no problem with this pull request, could you merge it, please?

**Comment by ReeceStevens** [comment]
That is great news! Can’t wait to try it out. Thanks for the hard work getting this framework working with the beta tool chain :)

**Pull request closed without merging**